qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -109,6 +136,10 @@ def run(self, dag):
                      qubits.index(gate.qargs[1])))
         edges = set(self.coupling_map.get_edges())
 
+        if not self.strict_direction:
+            cxs = {frozenset(cx) for cx in cxs}
+            edges = {frozenset(edge) for edge in edges}
```

Comment:
Can you open a retworkx issue here on this https://github.com/Qiskit/retworkx/issues/new/choose I feel like there should be a native method or function on the `PyDiGraph` class that` CouplingMap` wraps to return the set of weakly connected edges for a graph. It'd be a bit faster too if that method exists (you might be able to hack it together inside couplingmap today with `.to_undirected` https://qiskit.org/documentation/retworkx/stubs/retworkx.PyDiGraph.to_undirected.html#retworkx.PyDiGraph.to_undirected but having a native method would be best.